apixu-weatherbot1.0.tcl

```diff
@@ -0,0 +1,498 @@
+        # Looks up userinfo in the XTRA of userfile.
+        #
+        # Args:
+        #   nick(string) - Users nick.
+        #   hand(string) - Users handle.
+        #
+        # Returns:
+        #   (dict) - Returns userinfo or error if user not found.
+        #
+        putlog "weather::_getuserinfo looking up user location and units"
+        set location [getuser $hand XTRA weather.location]
+        set units [getuser $hand XTRA weather.units]
+        set userinfo [dict create location $location units $units]
+
+        if {![string length [dict get $userinfo location]]} {
+            putlog "weather::_getusuerinfo did not find info set for $nick"            
+            return [dict create error "Did not find a user"]
+        }
+ 
+       return $userinfo 
+    }
+
+    proc _get_json {location type} {
+        # Fetches JSON data from the apixu API.
+        #
+        # Args:
+        #   location(string) - Users location given.
+        #   type(string) - Current weather or forecast called.
+        #
+        # Returns:
+        #   (string) Returns json data received from the api.
+        #
+        putlog "weather::_get_json was called"
+
+        # First regexp matches any location given by user that is not
+        # a-z, A-Z, number, period, comma or space. This is sanitize
+        # the location string given to catch any illegal characters
+        # before the API call that will return an error.
+        #
+        # Second regexp matches any location give by user that starts
+        # and ends with a-z or A-Z with a : in the between the two. This
+        # is to let strings like "iata:ksjc" to be passed through as the
+        # only illegal character if used the correct way.
+        #
+        # If the first regexp is true and second regexp is false, the
+        # conditional will be true and throw an error for illegal characters.
+        if {[regexp {[^\w., ]} $location] && ![regexp {^[a-zA-Z]+:[a-zA-Z]+$} $location]} {
+            putlog "weather::_get_json caught illegal characters"
+            error "Illegal characters in your location query. Acceptable characters:\
+                   a-z, 0-9, commas, and spaces. Colons only used i.e. 'iata:ksjc'"
+        }
+
+        # Replace all spaces in $location with %20 for theapi link call
+        regsub -all -- { } $location {%20} location
+
+        set url "$::weather::base_url/$type.json?key=$::weather::apikey&q=$location"
+        if {$type ne "current"} {
+            set url "$::weather::base_url/$type.json?key=$::weather::apikey&q=$location&days=5"
+        }
+
+        putlog "weather::_get_json getting data from $url"
+        if {[catch {set token [::http::geturl $url -timeout 10000]} errormsg]} {
+            putlog "::http::geturl Error: $errormsg"
+            error "No matching location found"
+        }
+
+        set data [::http::data $token]
+        set status [::http::status $token]
+        set ncode [::http::ncode $token]
```

Comment:
`ncode` isn't very obvious to me... how about `response_code`?  characters are free.

Reply:
Good point. I agree, bad naming.